src/v2/ion/ui-schema/ion-object-handler.js

```diff
@@ -65,8 +65,28 @@ export function createOVOptions(options = []) {
   }
 }
 
+/**
+ * Example of the option like
+ * @param {AuthenticatorOption[]} options
+ * @param {( AuthenticatorEnrollment[] || Authenticator[] )} authenticators
+ */
+function updateCustomAppOptions(options = []) {
+  // update custom app options such that methodType value is sent to backend.
+  const customAppItems = options.filter((option) => option.relatesTo.key === AUTHENTICATOR_KEY.CUSTOM_APP);
+  customAppItems.forEach((customAppItem) => {
+    const methodTypeObj = customAppItem?.value?.form?.value?.find(v => v.name === 'methodType');
+    if (methodTypeObj) {
+      // set the methodType field to be required and immutable in our UI,
+      // so it is always sent to the backend.
+      methodTypeObj.required = true;
```

Comment:
Can you explain why this is necessary? Why isn't the backend setting `methodType` as `required`?

Reply:
@jaredperreault-okta So this is the [backend method ](https://github.com/okta/okta-core/blob/master/components/platformx/idx/impl/src/main/java/com/okta/api/mediation/services/auth/IdxRemediationBuilderImpl.java#L1461)which sets required as true or false based on requiresMethodTypeForVerification as this has to support initial challenge this needs to send required: false all the time.

Slack thread when they made the [same change for Okta Verify in UI](https://github.com/okta/okta-signin-widget/blob/e34474f134a4c4685dd48fa368a4d2b724352256/src/v2/ion/ui-schema/ion-object-handler.js#L39)
https://okta.slack.com/archives/C8WSD3TM4/p1597760218289000